pkg/proctree/proctree.go

```diff
@@ -39,8 +39,31 @@ const (
 	DefaultThreadCacheSize  = 32768
 )
 
+type SourceType int
+
+const (
+	SourceNone    SourceType = iota // disabled
+	SourceSignals                   // event from control plane enrich the process tree
+	SourceEvents                    // event from pipeline enrich the process tree
+	SourceBoth                      // events from both pipelines enrich the process tree
+)
+
+func (s SourceType) String() string {
+	switch s {
+	case SourceNone:
+		return "none"
+	case SourceSignals:
+		return "signals"
+	case SourceEvents:
+		return "events"
+	case SourceBoth:
+		return "signals and events"
+	}
+	return "unknown"
+}
+
 type ProcTreeConfig struct {
-	Enabled          bool
+	Source           SourceType
```

Comment:
I don't agree with putting this config here.
The source of the process tree is irrelevant, as you are not using it anywhere.
In the end, it determines how the object which has reference to the process tree (`Tracee`) is feeding it, which is not touching the process tree logic at all.
You should export it to the general config.

Reply:
Yep but then I would create a single item in the general config, related to the process tree, and apart from an already existing struct configuring the tree. No right or wrong.

Reply:
I am for having one config for the tree in the `proctree` package, and one in the general config.
You then only need to translate what you are configured with to the `proctree` config.
This ensures the independence of both, so each only includes what is relevant for itself.

Reply:
I removed the config variable from inside proctree (so its only dedicated to the singleton Config).